lib/rspec/core/memoized_helpers.rb

```diff
@@ -153,6 +153,17 @@ def initialize
 
         def fetch_or_store(key)
           @memoized.fetch(key) do # only first access pays for synchronization
+            if @mutex.locked? && !@mutex.owned?
+              raise <<-EOS
+Possible deadlock at:
+  #{CallerFilter.first_non_rspec_line}
+
+This is normally caused when accessing an uninitialized `let` value in a thread
+different than the one calling `subject`. Either use `let!` or disable RSpec
```

Comment:
+1 on the reasoning
How about this for a message?

> This is caused when one let block is waiting for a thread that accesses another `let` block. The thread gets put to sleep until the first let block finishes, but it never will since it is waiting for the thread.